{
  "artifacts": [],
  "command": "my_main",
  "experiment": {
    "base_dir": "/Users/jonnycook/Desktop/MARLGen/epymarl/src",
    "dependencies": [
      "munch==2.5.0",
      "numpy==1.21.5",
      "PyYAML==6.0",
      "sacred==0.8.2",
      "torch==1.10.2"
    ],
    "mainfile": "main.py",
    "name": "pymarl",
    "repositories": [],
    "sources": [
      [
        "main.py",
        "_sources/main_55bde9aaedfb0174d6e6bcc3ff03aada.py"
      ],
      [
        "run.py",
        "_sources/run_bd291ddbed9b8412db0ed2c1a032e31b.py"
      ],
      [
        "utils/logging.py",
        "_sources/logging_8309e899ab76a4e610428ba7ee8d8acb.py"
      ]
    ]
  },
  "fail_trace": [
    "Traceback (most recent call last):\n",
    "  File \"/Users/jonnycook/opt/anaconda3/envs/MARLGen/lib/python3.7/site-packages/sacred/config/captured_function.py\", line 42, in captured_function\n    result = wrapped(*args, **kwargs)\n",
    "  File \"src/main.py\", line 54, in my_main\n    run(_run, config, _log)\n",
    "  File \"/Users/jonnycook/Desktop/MARLGen/epymarl/src/run.py\", line 57, in run\n    run_sequential(args=args, logger=logger)\n",
    "  File \"/Users/jonnycook/Desktop/MARLGen/epymarl/src/run.py\", line 185, in run_sequential\n    episode_batch = runner.run(test_mode=False)\n",
    "  File \"/Users/jonnycook/Desktop/MARLGen/epymarl/src/runners/episode_runner.py\", line 78, in run\n    actions = self.mac.select_actions(self.batch, t_ep=self.t, t_env=self.t_env, test_mode=test_mode)\n",
    "  File \"/Users/jonnycook/Desktop/MARLGen/epymarl/src/controllers/basic_controller.py\", line 26, in select_actions\n    chosen_actions = self.action_selector.select_action(agent_outputs[bs], avail_actions[bs], t_env, test_mode=test_mode)\n",
    "  File \"/Users/jonnycook/Desktop/MARLGen/epymarl/src/components/action_selectors.py\", line 81, in select_action\n    m = Categorical(agent_inputs)\n",
    "  File \"/Users/jonnycook/opt/anaconda3/envs/MARLGen/lib/python3.7/site-packages/torch/distributions/categorical.py\", line 64, in __init__\n    super(Categorical, self).__init__(batch_shape, validate_args=validate_args)\n",
    "  File \"/Users/jonnycook/opt/anaconda3/envs/MARLGen/lib/python3.7/site-packages/torch/distributions/distribution.py\", line 56, in __init__\n    f\"Expected parameter {param} \"\n",
    "ValueError: Expected parameter probs (Tensor of shape (1, 2, 5)) of distribution Categorical(probs: torch.Size([1, 2, 5])) to satisfy the constraint Simplex(), but found invalid values:\ntensor([[[ 0.5545,  0.0120, -0.1896,  0.2742,  0.3488],\n         [ 0.4937,  0.1016, -0.1188,  0.2707,  0.2529]]],\n       grad_fn=<DivBackward0>)\n"
  ],
  "heartbeat": "2022-07-15T12:47:23.280380",
  "host": {
    "ENV": {},
    "cpu": "Apple M1",
    "hostname": "Jonnys-MBP",
    "os": [
      "Darwin",
      "Darwin-20.5.0-x86_64-i386-64bit"
    ],
    "python_version": "3.7.11"
  },
  "meta": {
    "command": "my_main",
    "options": {
      "--beat-interval": null,
      "--capture": null,
      "--comment": null,
      "--debug": false,
      "--enforce_clean": false,
      "--file_storage": null,
      "--force": false,
      "--help": false,
      "--loglevel": null,
      "--mongo_db": null,
      "--name": null,
      "--pdb": false,
      "--print-config": false,
      "--priority": null,
      "--queue": false,
      "--s3": null,
      "--sql": null,
      "--tiny_db": null,
      "--unobserved": false,
      "COMMAND": null,
      "UPDATE": [],
      "help": false,
      "with": false
    }
  },
  "resources": [],
  "result": null,
  "start_time": "2022-07-15T12:47:22.968260",
  "status": "FAILED",
  "stop_time": "2022-07-15T12:47:23.281221"
}